### PR TITLE
miscellaneous fixes for test execution

### PR DESCRIPTION
**Pull request by myronkscott**

### CODE REVIEW

File: common/src/main/java/com/tc/lang/TCThreadGroup.java [hunk 1]
```diff
@@ -105,6 +106,24 @@ public boolean retire(long timeout, Consumer<InterruptedException> interruptHand
     return complete;
   }
 
+  private boolean lookForThreadExit(Thread t, Consumer<InterruptedException> interruptHandler) {
+    if (ignorePoolThreads && (t.getName().startsWith("pool-") || t.getName().startsWith("ForkJoinPool.commonPool-worker-"))) {
```
Comment (chrisdennis) [inline common/src/main/java/com/tc/lang/TCThreadGroup.java:110]: `t instanceof ForkJoinWorkerThread` would at least make this slightly less ugly.

File: common/src/main/java/com/tc/util/version/VersionCompatibility.java [hunk 2]
```diff
@@ -47,6 +47,20 @@ private static boolean isCompatible(Version v1, Version v2) {
     return ((v1.major() == v2.major()) && (v1.minor() == v2.minor()));
   }
 
+  public static boolean isNewer(Version v1, Version v2) {
+    if (v1 == null || v2 == null) { throw new NullPointerException(); }
+    if (v1.major() > v2.major()) {
```
Comment (chrisdennis) [inline common/src/main/java/com/tc/util/version/VersionCompatibility.java:52]: `return v1.compareTo(v2) > 0`

**Review by chrisdennis — CHANGES_REQUESTED** [review]

File: common/src/main/java/com/tc/lang/TCThreadGroup.java (hunk 1 above)
Comment (myronkscott) [reply common/src/main/java/com/tc/lang/TCThreadGroup.java:110]: that would mean any fork join worker thread.  not just the common pool.

File: common/src/main/java/com/tc/util/version/VersionCompatibility.java (hunk 2 above)
Comment (myronkscott) [reply common/src/main/java/com/tc/util/version/VersionCompatibility.java:52]: that goes very deep including patch and build number.  I don't think we want to go that deep with this.

File: common/src/main/java/com/tc/lang/TCThreadGroup.java (hunk 1 above)
Comment (chrisdennis) [reply common/src/main/java/com/tc/lang/TCThreadGroup.java:110]: Sorry...:
```
(t instanceof ForkJoinWorkerThread) && ((ForkJoinWorkerThread) t).getPool() == ForkJoinPool.commonPool()
```

**Comment by chrisdennis** [comment]
@myronkscott I approved... let me know if you want me to merge.

**Pull request merged**